icubam/config.py

```diff
@@ -47,13 +47,29 @@ def _preprocess(self, conf):
         result[k.lower()] = v
       else:
         result[k.lower()] = self._preprocess(v)
-
     return result
 
   def __getitem__(self, key: str):
     if key.upper() == key:
       return self.env.get(key)
-    return self.conf.get(key) if self.conf.has_key(key) else None
+    if self.conf.has_key(key):
+      return self.conf.get(key)
+    raise KeyError(f"key '{key}' does not exist!")
```

Comment:
I think the original motivation to return None was to fix https://github.com/icubam/icubam/issues/327 however it doesn't fix it since the issue is in the dotmap library (cf my comment there).

Reply:
the problem is not fixed indeed. it is not so easy and would probably need to remove DotMap (or to inherit from it ?)
Because when we do: self.config.a.b.c, it does self.config.a first, this is the responsibility of the Config class. But the getter returns a DotMap object, lets call it dm. Then it is dm.b and it has now nothing to do with the config class any longer, but only has to do with DotMap. Lets' add a TODO for now.